lib/restaurant.rb

```diff
@@ -0,0 +1,48 @@
+#Understands a customers order
+require_relative 'menu'
+require_relative 'sms'
+
+
+class Restaurant
+
+  attr_reader :basket, :total, :sms
+
+  def initialize(menu = Menu.new, sms = Sms.new)
+    @menu = menu
+    @basket = {}
+    @total = 0
+    @sms = sms
+  end
+
+  def print_menu
+    puts
+    puts " --- Welcome to Mr Shakes---".center(25)
+    puts
+    puts "Please choose from our menu:".center(25)
+    puts
+    @menu.dishes.each do |dish, price|
+      puts "#{dish.ljust(10)} £#{price}".rjust(20)
+    end
+  end
+
+  def add_order(item, quantity=1)
+    @menu.dish_on_menu(item)
+    @basket[item] = quantity
+    @total += @menu.price(item)*quantity
+  end
+
+  def check_order
+    puts "Here's a summary of your order:".center(25)
```

Comment:
too many responsibilities - is it checking the order, or compiling a message for some user?